nomad/structs/structs.go

```diff
@@ -7672,6 +7672,11 @@ func (t *Task) MakeUniqueIdentityName(taskGroup string) string {
 	return fmt.Sprintf("%v-%v", taskGroup, t.Name)
 }
 
+// GetConsulTaskName returns the consulClusterName_TaskName string.
+func (t *Task) GetConsulTaskName() string {
+	return fmt.Sprintf("%s_%s", t.Consul.Cluster, t.Name)
+}
+
```

Comment:
We don't need the task name here, just the cluster. 

I think this got confusing in our discussion because at some point I used a method in my example. The reason I did that was because the `consul` block could be either at the group or task level, so we would need a function to pick the right cluster value. 

But I think @tgross mentioned he would work on that? For this PR I think you can just assume `task.Consul.Cluster` has what you need. Later we could change it to something `task.GetConsul().Cluster`, where `GetConsul()` would handle resolving the expected Consul config.

Reply:
But that would mean users could only have 1 templates Consul token per jobspec?

Reply:
"But that would mean users could only have 1 templates Consul token per ~jobspec~ task?"

This function would be used in the context of a specific task, and in this context we can only have one identity called `consul_<cluster>`.

Look at it from the jobspec perspective:

```hcl
job "..." {
  group"a" {
    task "1" {
      identity {
        name = "consul_default"
      }
    }

    task "2" {
      identity {
        name = "consul_default"
      }
    }

  group "b" {
    task "1" {
      identity {
        name = "consul_default"
      }
    }

    task "2" {
      identity {
        name = "consul_default"
      }
    }

    task "3" {
      identity {
        name = "consul_default"
      }
    }
  }
}
```

Despite all tasks having an identity called `consul_default` they are all different because they are placed in different tasks.